Error "Can't resolve 'crypto' in node_modules\@azure\storage-blob\dist-esm\storage-blob\src\credentials

- **Package Name**: @azure/storage-blob
- **Package Version**: 12.4.0
- **Operating system**: Windows 10
- [x] **typescript** 
    - **version**: 4.0.5

**Describe the bug**
After upgrading  from "@azure/storage-blob": "^12.2.1", to  "@azure/storage-blob": "^12.4.0", the Angular 11 compiler shows following error after 'ng serve' was executed:
`
Error: ./node_modules/@azure/storage-blob/dist-esm/storage-blob/src/credentials/UserDelegationKeyCredential.js
Module not found: Error: Can't resolve 'crypto' in 'C:\Projects\Angular-Portal\node_modules\@azure\storage-blob\dist-esm\storage-blob\src\credentials'

Error from chokidar (C:\): Error: EBUSY: resource busy or locked, lstat 'C:\DumpStack.log.tmp'
Error from chokidar (C:\): Error: EBUSY: resource busy or locked, lstat 'C:\hiberfil.sys'
Error from chokidar (C:\): Error: EBUSY: resource busy or locked, lstat 'C:\pagefile.sys'
Error from chokidar (C:\): Error: EBUSY: resource busy or locked, lstat 'C:\swapfile.sys'
`

**To Reproduce**
Steps to reproduce the behavior:
1. use new package with Angular 11 and Typescript 4.0.5

**Expected behavior**
No error and working sdk.

**Additional context**
With previous version the same code is working. 
"@azure/identity": "^1.1.0" was upgraded to "@azure/identity": "^1.2.2" without any issues. Is also working with "@azure/storage-blob": "^12.2.1".

I have the same issue with 12.4.0. Version 12.3.0 works fine for me.

Here's the error log:
```
ERROR in ./node_modules/@azure/storage-blob/dist-esm/storage-blob/src/credentials/UserDelegationKeyCredential.js
Module not found: Error: Can't resolve 'crypto' in '/Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/dist-esm/storage-blob/src/credentials'
resolve 'crypto' in '/Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/dist-esm/storage-blob/src/credentials'
  Parsed request is a module
  using description file: /Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/package.json (relative path: ./dist-esm/storage-blob/src/credentials)
    resolve as module
      looking for modules in /Users/build/Source/SampleProject/client-app
        using description file: /Users/build/Source/SampleProject/client-app/package.json (relative path: .)
          Field 'browser' doesn't contain a valid alias configuration
          using description file: /Users/build/Source/SampleProject/client-app/package.json (relative path: ./crypto)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/build/Source/SampleProject/client-app/crypto doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/build/Source/SampleProject/client-app/crypto.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /Users/build/Source/SampleProject/client-app/crypto.tsx doesn't exist
            .mjs
              Field 'browser' doesn't contain a valid alias configuration
              /Users/build/Source/SampleProject/client-app/crypto.mjs doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/build/Source/SampleProject/client-app/crypto.js doesn't exist
            as directory
              /Users/build/Source/SampleProject/client-app/crypto doesn't exist
      /Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/dist-esm/storage-blob/src/credentials/node_modules doesn't exist or is not a directory
      /Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/dist-esm/storage-blob/src/node_modules doesn't exist or is not a directory
      /Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/dist-esm/storage-blob/node_modules doesn't exist or is not a directory
      /Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/dist-esm/node_modules doesn't exist or is not a directory
      /Users/build/Source/SampleProject/client-app/node_modules/@azure/node_modules doesn't exist or is not a directory
      /Users/build/Source/SampleProject/client-app/node_modules/node_modules doesn't exist or is not a directory
      /Users/build/Source/node_modules doesn't exist or is not a directory
      /Users/build/node_modules doesn't exist or is not a directory
      /Users/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory
      looking for modules in /Users/build/Source/SampleProject/client-app/node_modules
        using description file: /Users/build/Source/SampleProject/client-app/package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
          using description file: /Users/build/Source/SampleProject/client-app/package.json (relative path: ./node_modules/crypto)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
      looking for modules in /Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/node_modules
        using description file: /Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/package.json (relative path: ./node_modules)
          using description file: /Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/package.json (relative path: ./node_modules/crypto)
            no extension
              /Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/node_modules/crypto doesn't exist
              /Users/build/Source/SampleProject/client-app/node_modules/crypto doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
            .ts
              /Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/node_modules/crypto.ts doesn't exist
              /Users/build/Source/SampleProject/client-app/node_modules/crypto.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
      looking for modules in /Users/build/Source/SampleProject/node_modules
        No description file found
            .tsx
              /Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/node_modules/crypto.tsx doesn't exist
              /Users/build/Source/SampleProject/client-app/node_modules/crypto.tsx doesn't exist
            .mjs
              Field 'browser' doesn't contain a valid alias configuration
            .mjs
              /Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/node_modules/crypto.mjs doesn't exist
              /Users/build/Source/SampleProject/client-app/node_modules/crypto.mjs doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
            .js
              /Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/node_modules/crypto.js doesn't exist
            as directory
              /Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/node_modules/crypto doesn't exist
              /Users/build/Source/SampleProject/client-app/node_modules/crypto.js doesn't exist
            as directory
              /Users/build/Source/SampleProject/client-app/node_modules/crypto doesn't exist
        No description file found
        no extension
          /Users/build/Source/SampleProject/node_modules/crypto doesn't exist
        .ts
          /Users/build/Source/SampleProject/node_modules/crypto.ts doesn't exist
        .tsx
          /Users/build/Source/SampleProject/node_modules/crypto.tsx doesn't exist
        .mjs
          /Users/build/Source/SampleProject/node_modules/crypto.mjs doesn't exist
        .js
          /Users/build/Source/SampleProject/node_modules/crypto.js doesn't exist
        as directory
          /Users/build/Source/SampleProject/node_modules/crypto doesn't exist
[/Users/build/Source/SampleProject/client-app/crypto]
[/Users/build/Source/SampleProject/client-app/crypto.ts]
[/Users/build/Source/SampleProject/client-app/crypto.tsx]
[/Users/build/Source/SampleProject/client-app/crypto.mjs]
[/Users/build/Source/SampleProject/client-app/crypto.js]
[/Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/dist-esm/storage-blob/src/credentials/node_modules]
[/Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/dist-esm/storage-blob/src/node_modules]
[/Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/dist-esm/storage-blob/node_modules]
[/Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/dist-esm/node_modules]
[/Users/build/Source/SampleProject/client-app/node_modules/@azure/node_modules]
[/Users/build/Source/SampleProject/client-app/node_modules/node_modules]
[/Users/build/Source/node_modules]
[/Users/build/node_modules]
[/Users/node_modules]
[/node_modules]
[/Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/node_modules/crypto]
[/Users/build/Source/SampleProject/client-app/node_modules/crypto]
[/Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/node_modules/crypto.ts]
[/Users/build/Source/SampleProject/client-app/node_modules/crypto.ts]
[/Users/build/Source/SampleProject/node_modules/package.json]
[/Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/node_modules/crypto.tsx]
[/Users/build/Source/SampleProject/client-app/node_modules/crypto.tsx]
[/Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/node_modules/crypto.mjs]
[/Users/build/Source/SampleProject/client-app/node_modules/crypto.mjs]
[/Users/build/Source/SampleProject/client-app/node_modules/@azure/storage-blob/node_modules/crypto.js]
[/Users/build/Source/SampleProject/client-app/node_modules/crypto.js]
[/Users/build/Source/SampleProject/node_modules/crypto/package.json]
[/Users/build/Source/SampleProject/node_modules/crypto]
[/Users/build/Source/SampleProject/node_modules/crypto.ts]
[/Users/build/Source/SampleProject/node_modules/crypto.tsx]
[/Users/build/Source/SampleProject/node_modules/crypto.mjs]
[/Users/build/Source/SampleProject/node_modules/crypto.js]
 @ ./node_modules/@azure/storage-blob/dist-esm/storage-blob/src/credentials/UserDelegationKeyCredential.js 1:0-36 32:15-25
 @ ./node_modules/@azure/storage-blob/dist-esm/storage-blob/src/sas/BlobSASSignatureValues.js
 @ ./node_modules/@azure/storage-blob/dist-esm/storage-blob/src/Clients.js
 @ ./node_modules/@azure/storage-blob/dist-esm/storage-blob/src/index.browser.js
 @ ./src/app/services/sas-store.ts
 @ ./src/app/components/announcement-manager/create-announcement/create-announcement.component.ts
 @ ./src/app/components/announcement-manager/announcement/announcement.component.ts
 @ ./src/app/components/announcement-manager/announcement-manager.component.ts
 @ ./src/app/app-routing.module.ts
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi ./src/main.ts
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! obc-portal@0.0.0 build: `ng build --prod`
npm ERR! Exit status 1
```

--- a/sdk/storage/storage-blob/CHANGELOG.md
+++ b/sdk/storage/storage-blob/CHANGELOG.md
@@ -1,5 +1,9 @@
 # Release History
 
+## 12.4.1 (Unreleased)
+
+- Fixed a compile failure due to "Can't resolve 'crypto'" in Angular. [Issue #13267](https://github.com/Azure/azure-sdk-for-js/issues/13267).
+
 ## 12.4.0 (2021-01-12)
 
 - Added a new `from(permissionLike)` function to `AccountSASPermissions`, `BlobSASPermissions` and `ContainerSASPermissions` for creating such a permission from a raw permission-like object. Addressed issue [9714](https://github.com/Azure/azure-sdk-for-js/issues/9714).

--- a/sdk/storage/storage-blob/package.json
+++ b/sdk/storage/storage-blob/package.json
@@ -8,6 +8,7 @@
   "browser": {
     "./dist-esm/storage-blob/src/index.js": "./dist-esm/storage-blob/src/index.browser.js",
     "./dist-esm/storage-blob/src/credentials/StorageSharedKeyCredential.js": "./dist-esm/storage-blob/src/credentials/StorageSharedKeyCredential.browser.js",
+    "./dist-esm/storage-blob/src/credentials/UserDelegationKeyCredential.js": "./dist-esm/storage-blob/src/credentials/UserDelegationKeyCredential.browser.js",
     "./dist-esm/storage-blob/src/utils/utils.node.js": "./dist-esm/storage-blob/src/utils/utils.browser.js",
     "./dist-esm/storage-blob/test/utils/index.js": "./dist-esm/storage-blob/test/utils/index.browser.js",
     "./dist-esm/storage-blob/src/BatchUtils.js": "./dist-esm/storage-blob/src/BatchUtils.browser.js",

--- a/sdk/storage/storage-blob/src/credentials/UserDelegationKeyCredential.browser.ts
+++ b/sdk/storage/storage-blob/src/credentials/UserDelegationKeyCredential.browser.ts
@@ -1,0 +1,4 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+export class UserDelegationKeyCredential {}

--- a/sdk/storage/storage-file-datalake/CHANGELOG.md
+++ b/sdk/storage/storage-file-datalake/CHANGELOG.md
@@ -1,5 +1,9 @@
 # Release History
 
+## 12.3.1 (Unreleased)
+
+- Fixed a compile failure due to "Can't resolve 'crypto'" in Angular. [Issue #13267](https://github.com/Azure/azure-sdk-for-js/issues/13267).
+
 ## 12.3.0 (2021-01-12)
 
 - Bug fix - `DataLakePathClient.move()` now supports source and destination authenticated with SAS. Fixed bug [12758](https://github.com/Azure/azure-sdk-for-js/issues/12758).

--- a/sdk/storage/storage-file-datalake/package.json
+++ b/sdk/storage/storage-file-datalake/package.json
@@ -16,6 +16,7 @@
   "browser": {
     "./dist-esm/storage-file-datalake/src/index.js": "./dist-esm/storage-file-datalake/src/index.browser.js",
     "./dist-esm/storage-file-datalake/src/credentials/StorageSharedKeyCredential.js": "./dist-esm/storage-file-datalake/src/credentials/StorageSharedKeyCredential.browser.js",
+    "./dist-esm/storage-file-datalake/src/credentials/UserDelegationKeyCredential.js": "./dist-esm/storage-file-datalake/src/credentials/UserDelegationKeyCredential.browser.js",
     "./dist-esm/storage-file-datalake/src/utils/utils.node.js": "./dist-esm/storage-file-datalake/src/utils/utils.browser.js",
     "./dist-esm/storage-file-datalake/test/utils/index.js": "./dist-esm/storage-file-datalake/test/utils/index.browser.js",
     "./dist-esm/storage-common/src/index.js": "./dist-esm/storage-common/src/index.browser.js",

--- a/sdk/storage/storage-file-datalake/src/credentials/UserDelegationKeyCredential.browser.ts
+++ b/sdk/storage/storage-file-datalake/src/credentials/UserDelegationKeyCredential.browser.ts
@@ -1,0 +1,4 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+export class UserDelegationKeyCredential {}

